src/ray/core_worker/lease_policy.cc

```diff
@@ -26,6 +26,11 @@ std::pair<rpc::Address, bool> LocalityAwareLeasePolicy::GetBestNodeForTask(
     return std::make_pair(fallback_rpc_address_, false);
   }
 
+  if (spec.GetMessage().scheduling_strategy().scheduling_strategy_case() ==
+      rpc::SchedulingStrategy::SchedulingStrategyCase::kNodeSchedulingStrategy) {
+    return std::make_pair(fallback_rpc_address_, false);
```

Comment:
Could we return the node ID specified in this case? That way we can skip the first round of lease requests to the local raylet.

Reply:
Also, how does this work for actors? Does the GCS choose a random node?

Reply:
Yes, I can make that optimization. Actor can apply the same optimization but it should also work without it since the initial raylet will redirect to the correct specified raylet.

Reply:
yeah this is purely an optimization.